forum/qa-plugin/user-activity-log/user-activity-log-search.php

```diff
@@ -0,0 +1,177 @@
+        }else if($this->searchArr['condition'] == 'type'){
+            $finalQuery = $baseQuery.'WHERE event = $';
+        }else if($this->searchArr['condition'] == 'ip'){
+            if(qa_get_logged_in_level() > QA_USER_LEVEL_EDITOR){
+                $finalQuery = $baseQuery.'WHERE ipaddress = $';
+            }   
+        }
+
+        if($this->searchArr['date'] != ""){
+            $dateArr = explode('-', $this->searchArr['date']);
+
+
+            if(is_numeric($dateArr[0]) && is_numeric($dateArr[0])){
+                $date = $dateArr[0];
+                if(isset($dateArr[1]) && is_numeric($dateArr[1])){
+                    $date = $date.'-'.$dateArr[1];
+                    if(isset($dateArr[2]) && is_numeric($dateArr[2])){
+                        if(strlen($dateArr[2]) > 2){
+                            $time = explode(' ', $dateArr[2]);
+                            $dateArr[2] = $time[0];
+                            $dateArr[3] = $time[1];
+                        }
+                        
+                        $date = $date.'-'.$dateArr[2];
+                    }
+                }
+                $finalQuery = $finalQuery.'AND `datetime` LIKE $';
+            }
+
+        }
+        
+        
+        if(!is_numeric($this->searchArr['resultsCount'])){
+            $this->dbResult = [];
+            return 0;
+        }
+        
+        $finalQuery = $finalQuery.' ORDER BY `datetime` DESC LIMIT #';
+        if(strpos($finalQuery, 'LIKE')){
+            $result = qa_db_query_sub($finalQuery, $this->searchArr['request'], $this->searchArr['date']."%", $this->searchArr['resultsCount']);
+        }else{
+            $result = qa_db_query_sub($finalQuery, $this->searchArr['request'], $this->searchArr['resultsCount']);
+        }
+       
+        if(mysqli_num_rows($result) <= $this->searchArr['resultsCount']){
+            $fullResultsArray = [];
+            while($row = mysqli_fetch_array($result, MYSQLI_ASSOC)){
+                $fullResultsArray[] = $row;
+            }
+            $this->dbResult = $fullResultsArray;
+            return 1;
+        }else{
+            $this->dbResult = $result;
+            return 0;
+        }
+
+    }
+
+    private function generateResultsTable()
+    {
+        $unfiltered = $this->dbResult;
+        if(qa_get_logged_in_level() < QA_USER_LEVEL_ADMIN){
+           $results = array_filter($unfiltered, function($field){
+                switch($field['event']){
+                    case 'q_vote_up': return ""; break;
+                    case 'q_vote_down': return ""; break;
+                    case 'q_vote_nil': return ""; break;
+                    case 'a_vote_up':return ""; break;
+                    case 'a_vote_down': return ""; break;
+                    case 'a_vote_nil': return ""; break;
+                    case 'c_vote_up': return ""; break;
+                    case 'c_vote_down': return ""; break;
+                    case 'c_vote_nil': return ""; break;
```

Comment:
Ten powyższy zbiór `case`-ów chyba można spiąć w jeden blok? Żeby nie powtarzać `return ""; break;`.

Reply:
chodzi o coś w stylu `case 'c_vote_up', 'a_vote_nil' ... : return ""; break; `  czy tak bardziej jak jest to zrobione w liniach niżej?

Reply:
Tak, chodzi mi o użycie `return ""; break;` raz. :) Dodatkowo użyłbym klamer, żeby było bardziej czytelnie:
```php
case 'a_vote_nil':
case 'c_vote_up':
case 'c_vote_down':
case 'c_vote_nil': {
  return "";
  break;
}
```

Reply:
@ScriptyChris pierwszy raz widzę klamy w `case` :D

Reply:
Myślałem, że owijanie bloków w klamry to normalna rzecz w językach, gdzie... używa się klamr. :) [W JS-ie działa](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/switch#block-scope_variables_within_switch_statements) i [w PHP widzę, że też](jdoodle.com/ia/oE8). :P

Reply:
```php
case 'test': {
  $text = 'lorem ipsum';
  $sum = 1 + 2;

  echo($text);
  echo($sum);

  break;
}
```
vs
```php
case 'test':
  $text = 'lorem ipsum';
  $sum = 1 + 2;

  echo($text);
  echo($sum);

  break;
```

Wg mnie klamry dodają czytelności i jakbym miał porównać konstrukcję `case` do praktycznie każdego innego bloku (`if`, `for`, `while`, `function`, `try..catch`), to wg mnie spójnym jest, aby zawartość tych bloków owijać klamrami (może z wyjątkiem jedno-linijkowców, tam gdzie składnia pozwala na pomijanie klamer). :)

Reply:
Jak wolisz, nie ma tego doprecyzowanego w PSR-2 https://www.php-fig.org/psr/psr-2/#52-switch-case
Także PSR-12 nic o tym nie mówi: https://www.php-fig.org/psr/psr-12/#52-switch-case
Chcąc trzymać się standardu raczej zalecałbym odpuszczenie sobie tutaj klamer. Tak samo historii języka.

Reply:
> (może z wyjątkiem jedno-linijkowców, tam gdzie składnia pozwala na pomijanie klamer)

Ale to już jest zupełnie niezgodne ze standardami pisania w PHP.